fjord/base/l10n.py

```diff
@@ -32,3 +47,52 @@ def _parse_block(self, parser, allow_pluralize):
         parse_block = InternationalizationExtension._parse_block
         ref, buffer = parse_block(self, parser, allow_pluralize)
         return ref, strip_whitespace(buffer)
+
+def tweak_message(message):
+    """We piggyback on jinja2's babel_extract() (really, Babel's extract_*
+    functions) but they don't support some things we need so this function will
+    tweak the message.  Specifically:
+
+        1) We strip whitespace from the msgid.  Jinja2 will only strip
+            whitespace from the ends of a string so linebreaks show up in
+            your .po files still.
+
+        2) Babel doesn't support context (msgctxt).  We hack that in ourselves
+            here.
+    """
+    if isinstance(message, basestring):
+        message = strip_whitespace(message)
+    elif isinstance(message, tuple):
+        # A tuple of 2 has context, 3 is plural, 4 is plural with context
+        if len(message) == 2:
+            message = add_context(message[1], message[0])
+        elif len(message) == 3:
+            if all(isinstance(x, basestring) for x in message[:2]):
+                singular, plural, num = message
+                message = (strip_whitespace(singular),
+                           strip_whitespace(plural),
+                           num)
+        elif len(message) == 4:
+            singular, plural, num, ctxt = message
+            message = (add_context(ctxt, strip_whitespace(singular)),
+                       add_context(ctxt, strip_whitespace(plural)),
+                       num)
+    return message
```

Comment:
We did a check earlier and there are no instances of msgctxt, so we could nix the msgctxt stuff here. However, it's probably better to keep the transition simple for now so I think we should just leave it as is.